src/lib/process/index.ts

```diff
@@ -2,54 +2,74 @@ import { createNodeApi } from '../utils/polkadot.js'
 import { Constants } from './constants.js'
 import { createProcessTransaction, disableProcessTransaction, getVersion } from './api.js'
 import { utf8ToHex } from './hex.js'
-import { mapRestrictions } from './map.js'
-import type { ChainRestrictions } from '../types/restrictions.js'
-import { VersionError } from '../types/error.js'
+import { stepValidation } from '../types/restrictions.js'
+import { NoValidRestrictionsError, VersionError } from '../types/error.js'
 
 export const defaultOptions: Polkadot.Options = {
   API_HOST: 'localhost',
   API_PORT: 9944,
   USER_URI: '//Alice',
 }
 
+const validate = (program: Process.Program = []): Process.Program => {
+  return program.reduce((out: Process.Program, step: Process.ProgramStep)=> {
+    const validated: Process.ProgramStep = stepValidation.parse(step?.restriction || step)
+    if (Object.keys(validated).length === 0) {
+      return out
+    }
+    out.push(step) // .push in reduce is more effective than spread, can cause issuecs
+    return out
+  }, [])
+}
+
+
+export const loadProcesses = async ({ data, options, dryRun }: { data: string, options?: Polkadot.Options, dryRun?: boolean }): Promise<Process.Response|string> => {
+  try {
+    const res: Process.Response = {}
+    const processes: Process.CLIParsed = JSON.parse(data)
+    // TODO more elegant promise.series way.
+    for (let i = 0; i < processes.length; i++) {
+      const { name, version, program } = processes[i]
+      res[name] = await createProcess(name, version, program, dryRun, options)
+    }
+
+    return res
+  } catch (err) {
+    // TODO handle error
+    return `Error occured: ${err}`
```

Comment:
Should this not rethrow?

Reply:
well, now it does not really matter. The reason was to not wrap command in try/catch so it returns an error, however, i think this is no longer relevant so updating!

Reply:
updated, hope tests will pass:)